op-gateway/src/index.ts

```diff
@@ -1,25 +1,44 @@
 import { EVMGateway } from '@ensdomains/evm-gateway';
 import { JsonRpcProvider } from 'ethers';
 import { OPProofService, type OPProvableBlock } from './OPProofService.js';
+import {
+  OPDisputeGameProofService,
+  type OPProvableBlock as OPDisputeGameProvableBlock,
+} from './OPDisputeGameProofService.js';
 
-export type OPGateway = EVMGateway<OPProvableBlock>;
+export type OPGateway = EVMGateway<
+  OPProvableBlock | OPDisputeGameProvableBlock
+>;
 
 export async function makeOPGateway(
```

Comment:
Given that the third argument depends on the type of service - it's either an `L2OutputOracle` or a `DisputeGameFactory`, I'm not sure it makes sense to have this distinction here. Two separate functions might be clearer.